### PR TITLE
Enhanced multiselect dropdowns (SHOOP-2324)

### PR DESCRIPTION
**Pull request by tulimaki**
Admin: Add view and widget to support Select2 dropdowns
Admin: Enable multiselect dropdown with Select2
Admin: Remove support for selectpicker
Campaigns: Use Select2 for ManyToManyField's

### CODE REVIEW

**Comment by tulimaki** [comment]
retest this please

**Comment by suutari-ai** [comment]
Why support for select pickers was removed? (Commit message doesn't say anything.)

File: shoop/campaigns/forms.py [hunk 1]
```diff
@@ -77,13 +81,33 @@ def __init__(self, *args, **kwargs):
...
+                        formfield = field.formfield(**kwargs)
+                    self.fields[obj.identifier] = formfield
+
+    def _get_initial_value(self, obj):
+        initial_value = self.get_initial_value(obj.identifier)
+        if initial_value:
+            if not isinstance(initial_value, str):
+                initial_value = [o.pk for o in initial_value]
+            else:
+                initial_value = [initial_value]
```
Comment (suutari-ai) [inline shoop/campaigns/forms.py]: Got `TypeError: 'decimal.Decimal' object is not iterable` from the `initial_value = [o.pk for o in initial_value]` line of this.

**Comment by suutari-ai** [comment]
* IMHO Products should be selectable by SKU too (and probably SKU should be shown in the selected item and in the dropdown too)
* The select2 fields should be styled (especially since all select fields are now using the select2):
   * Focus color should be same as for other input fields.
   * Padding should be fixed to be same above and below the items.
   * Select 2 fields should have same weight as other fields (when they fit to single line)

File: shoop/campaigns/forms.py (hunk 1 above)
Comment (suutari-ai) [inline shoop/campaigns/forms.py]: How about this one instead?
```python
    def _get_initial_value(self, obj):
        initial_value = self.get_initial_value(obj.identifier)
        if initial_value is None:
            return None
        elif isinstance(initial_value, QuerySet):
            return [x.pk for x in initial_value]
        else:
            return [initial_value]
```

**Comment by tulimaki** [comment]
* Added better styles for select2 selects
* Added option to search with SKU.
* Changed object text to force_text(obj) instead of objecte name. I think it's up to Product __str__ to display sku. And I think it would add unneccesary complexety to decide display based on the object type.
* Added fix to _get_initial_value in campaign forms

**Pull request merged**